core/arch/disassemble_shared.c

```diff
@@ -607,10 +608,16 @@ internal_opnd_disassemble(char *buf, size_t bufsz, size_t *sofar INOUT,
         print_to_buffer(buf, bufsz, sofar, IF_X64("<re> ") "@" PFX "+%d",
                         opnd_get_instr(opnd), opnd_get_mem_instr_disp(opnd));
         break;
-    case REG_kind:
-        reg_disassemble(buf, bufsz, sofar, opnd_get_reg(opnd), opnd_get_flags(opnd), "",
-                        "");
+    case REG_kind: {
+        bool is_mask = could_be_mask && reg_is_opmask(opnd_get_reg(opnd));
```

Comment:
could_be_mask is really just there because internal_opnd_disassemble does not have the instr available. I wanted to pull in the decision to print {} into internal_opnd_disassemble as it belogns to the operand. The decision is based on the instruction (opmask instruction or not). So to get rid of could_be_mask, I would have ro print the {} outside of internal_opnd_disassemble, which I thought is not the right thing. WDYT?

Reply:
But the code here calls `reg_is_opmask` which sure looks like it's sufficient to know whether this is a mask??!?

Reply:
Ok yes so the entire point is to print {mask} for e.g. zmm1{k1} (EVEX mask).
But we don't want to print {} for scalar mask instructions like kmov k0, k1
This is actually how other disassemblers do it as well. The {} indicates that the operand is actually a mask. No {} means that it is a scalar.

Reply:
Hmm, that was not clear from the code, w/o being familiar with that.

This internal-only bool does not seem ideal.  Seems too specific to this one operand type (would prefer some kind of multi-info flags at that point, esp w/ another bool already there).

Possible alternatives:
+ Print these braces at the instruction level, not at the operand level.
+ Make this a different type of operand -- maybe too extreme.

Reply:
Seems like the name should be `is_mask_modifier` or sthg which is clearer with no ambiguity instead of `could_be_mask`.  But I would vote for printing at the instruction level: have instr disasm figure out whether to put braces around one of its operands.

Reply:
Done.